srcopsmetrics/kebechet_metrics.py

```diff
@@ -0,0 +1,227 @@
+        stats["rejected_by_other"] = stats["rejected"] - stats["rejected_by_kebechet_bot"]
+
+        stats["merged"] = len(prs) - stats["rejected"]
+        stats["merged_by_kebechet_bot"] = len(prs[prs["merged_by_kebechet_bot"] == 1])
+        stats["merged_by_other"] = stats["merged"] - stats["merged_by_kebechet_bot"]
+
+        return stats
+
+    def get_daily_stats_update_manager(self):
+        """Get daily stats.
+
+        If self.today set to true, return only stats for current day.
+        """
+        prs = self._get_update_manager_pull_requests()
+        prs["days"] = prs.apply(lambda x: datetime.fromtimestamp(x["date"]).date(), axis=1)
+
+        stats: Dict[datetime, Any] = {}
+        day_range = [datetime.now().date()] if self.today else prs["days"].unique()
+        for date in day_range:
+            prs_day = prs[prs["days"] == date]
+
+            day = {}
+            day["created_pull_requests"] = len(prs_day)
+
+            day["rejected"] = len(prs_day[np.isnan(prs_day["ttm"])])
+            day["rejected_by_kebechet_bot"] = len(prs_day[prs_day["rejected_by_kebechet_bot"] == 1])
+            day["rejected_by_other"] = day["rejected"] - day["rejected_by_kebechet_bot"]
+
+            day["merged"] = len(prs_day) - day["rejected"]
+            day["merged_by_kebechet_bot"] = len(prs_day[prs_day["merged_by_kebechet_bot"] == 1])
+            day["merged_by_other"] = day["merged"] - day["merged_by_kebechet_bot"]
+
+            if self.today:
+                return day
+
+            stats[str(date)] = day
+
+        return stats
+
+    def evaluate_and_store_kebechet_metrics(self, is_local: bool):
+        """Calculate and store metrics for every kebechet manager in repository."""
+        for get_stats in [self.update_manager]:
+            stats = get_stats()
+
+            path = f"./srcopsmetrics/metrics/{self.repo_name}/kebechet_{get_stats.__name__}"
+            if self.today:
+                curr_day = datetime.now().date()
+                path += f"_{str(curr_day)}"
+            path += ".json"
+
+            KnowledgeStorage(is_local=is_local).save_knowledge(file_path=Path(path), data=stats)
+
+    def update_manager(self):
+        """Calculate and store update manager metrics."""
+        overall_stats = self.get_overall_stats_update_manager()
+        daily_stats = self.get_daily_stats_update_manager()
+        return {"overall": overall_stats, "daily": daily_stats}
+
+    def label_bot_manager(self):
+        """Calculate and store label bot manager metrics."""
+        raise NotImplementedError
+
+    def thoth_advise(self):
+        """Calculate and store thoth advise manager metrics."""
+        raise NotImplementedError
+
+    def thoth_promenance(self):
+        """Calculate and store promenance manager metrics."""
+        raise NotImplementedError
+
+    def pipfile_requirements(self):
+        """Calculate and store pipfile requirements manager metrics."""
+        raise NotImplementedError
```

Comment:
@pacospace other managers

Reply:
Yes, just asking if you think we should separate them in different classes? :)

Reply:
Currently - i do not think so. All the managers' evaulations share prs and issues that are loaded with `mi` inside the `KebechetMetrics` class (see `__init__`).